src/node_file.cc

```diff
@@ -961,6 +962,30 @@ static void Open(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void CopyFile(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!args[0]->IsString())
+    return TYPE_ERROR("src must be a string");
```

Comment:
Since these are already being checked on the js side, perhaps just a `CHECK` in here?

Reply:
The JS layer checks don't actually verify this.

Reply:
ugh.. that's right. hmmm. we should definitely fix that.
would much prefer any new errors to go through the `internal/errors` path.

Reply:
I agree. `fs.js` in general needs some TLC.

Reply:
I'll see if I can schedule some time next week to work on that. The risk, of course, is that `fs` is one of the most-monkey-patched modules out there and we risk breaking a lot of people by being too aggressive.

Reply:
> fs is one of the most-monkey-patched modules out there and we risk breaking a lot of people by being too aggressive.

For new methods in `fs` it might then make sense to be more aggressive rather than not. Personally, the more assertions the better. Keeps things sharp.